src/vs/workbench/contrib/markers/browser/markers.contribution.ts

```diff
@@ -568,15 +568,34 @@ class MarkersStatusBarContributions extends Disposable implements IWorkbenchCont
 	) {
 		super();
 		this.markersStatusItem = this._register(this.statusbarService.addEntry(this.getMarkersItem(), 'status.problems', StatusbarAlignment.LEFT, 50 /* Medium Priority */));
-		this.markersStatusItemOff = this._register(this.statusbarService.addEntry(this.getMarkersItemTurnedOff(), 'status.problemsVisibility', StatusbarAlignment.LEFT, 49));
+
+		const addStatusBarEntry = () => {
+			this.markersStatusItemOff = this._register(this.statusbarService.addEntry(this.getMarkersItemTurnedOff(), 'status.problemsVisibility', StatusbarAlignment.LEFT, 49));
+		};
+
+		// Add the status bar entry if the problems is not visible
+		let config = this.configurationService.getValue('problems.visibility');
+		if (!config) {
+			addStatusBarEntry();
+		}
+
 		this._register(this.markerService.onMarkerChanged(() => {
 			this.markersStatusItem.update(this.getMarkersItem());
-			this.markersStatusItemOff.update(this.getMarkersItemTurnedOff());
+
 		}));
+
 		this._register(this.configurationService.onDidChangeConfiguration(e => {
```

Comment:
I think these listeners are going to be registered multiple times and not disposed, every time `addStatusBarEntry` runs

Reply:
removed`..._register.` to align with what i see in some other examples of this

Reply:
Oops I misread the github diff, I was talking about the listeners, but those are not inside `addStatusBarEntry`. But your change is correct.